File: library/src/main/java/io/ashdavies/rxtasks/RxTasks.java
package io.ashdavies.rxtasks;

import com.google.android.gms.tasks.Task;
import io.reactivex.Completable;
import io.reactivex.Single;

public class RxTasks {

  private RxTasks() {
    throw new IllegalStateException("No instances");
  }

  public static Completable completable(Task<Void> task) {
    return Completable.create(new CompletableTaskOnSubscribe(task));
  }

  public static <T> Single<T> single(final Task<T> task) {
    return Single.create(new SingleTaskOnSubscribe<>(task));
  }
}

File: library/src/main/kotlin/io/ashdavies/rx/rxtasks/RxTasks.kt
package io.ashdavies.rx.rxtasks

import com.google.android.gms.tasks.Task
import io.reactivex.Completable
import io.reactivex.Single
import java.util.concurrent.Executor

object RxTasks {

  @JvmStatic
  @JvmOverloads
  @Deprecated(
      message = "Use Task<Void>.toCompletable() instead",
      replaceWith = ReplaceWith(
          expression = "task.toCompletable()",
          imports = ["io.ashdavies.rx.rxtasks.toCompletable"]
      )
  )
  fun completable(task: Task<Void>, executor: Executor = MainThreadExecutor()): Completable = task.toCompletable(executor)

  @JvmStatic
  @JvmOverloads
  @Deprecated(
      message = "Use Task<T>.toSingle() instead",
      replaceWith = ReplaceWith(
          expression = "task.toSingle()",
          imports = ["io.ashdavies.rx.rxtasks.toSingle"]
      )
  )
  fun <T> single(task: Task<T>, executor: Executor = MainThreadExecutor()): Single<T> = task.toSingle(executor)
}
